Controller.cs

```diff
@@ -0,0 +1,63 @@
+using System;
+using System.Collections.Generic;
+using CarDrivers;
+
+/// <summary>
+/// The controller of the application. This class controls the flow of the computation between the evolution and evaluation phases of the genetic algorithm.
+/// </summary>
+public class Controller
+{
+    private static readonly Controller _instance = new Controller();
+
+    private static readonly int max_iter = 500;
+    private static readonly int genotypeDimension = 100;
+
+    private GeneticAlgorithm geneticAlgorithm;
+
+    private Controller() { }
+
+    /// <summary>
+    /// Returns the only instance of this class (Sigleton pattern).
+    /// </summary>
+    /// <returns>The instance of the controller.</returns>
+    public static Controller Instance 
+    {
+        get { return _instance; }
+    }
+
+    /// <summary>
+    /// Starts the computation, creating a random population of genotypes, and start the evaluation. <br/><br/>
+    /// Note, this method has to be called only once, at the start of the programme.
+    /// </summary>
+    /// <param name="populationSize">The size of the population (eg number of cars)</param>
+    public void Start(int populationSize) 
+    {
+        this.geneticAlgorithm = new GeneticAlgorithm(genotypeDimension, populationSize);
+        this.geneticAlgorithm.InitializePopulation();
+        RaceManager.Instance.AllCarsDead += this.CarEvolution;
+        this.CarEvolution();
+    }
+
+    private void CarEvolution() 
+    {
+        if (this.geneticAlgorithm.GenerationCount < max_iter)
+        {
+            if (this.geneticAlgorithm.GenerationCount > 1) // generation 1 must not be evolved!
+                this.geneticAlgorithm.Evolution();
+            RaceManager.Instance.SetupCars(this.CreateAgents(this.geneticAlgorithm.CurrentPopulation));
+            RaceManager.Instance.Restart();
+        }
+        else 
+        {
+            // end, something to do?
```

Comment:
I don't know if we have to do something if the evolution steps overstep the given max number of iterations. 
Maybe show an error message? 
I will edit this part if you ask @manuelperuzzi

Reply:
I think that showing a message could be the best solution, in order to explicit the end of the genetic algorithm execution. It's not a priority though, so we can postpone this task to a later time.